Components/SinglePokemon.tsx

```diff
@@ -14,43 +14,40 @@ import { useNavigation } from "@react-navigation/native";
 import { StackNavigationProp } from "@react-navigation/stack";
 import type { StackParamList } from "../Types/StackParams";
 
-interface Props {
-  name: string;
-}
 type favouritePokemonProp = StackNavigationProp<
   StackParamList,
   "FavouritePokemon"
 >;
 
-const convertToPokemonObject = (newName: string) => {
-  const pokemon: Pokemon = {
-    name: newName,
-  };
-  return pokemon;
-};
-
-export const SinglePokemon = ({ name }: Props) => {
+export const SinglePokemon = (pokemon: Pokemon) => {
   const [isFavourite, setIsFavourite] = useState<boolean>(false);
-  const { savePokemon, deletePokemon } = React.useContext(PokemonContext);
+  // eslint-disable-next-line operator-linebreak
+  const {
+    addPokemonToFavourites,
+    deletePokemonFromFavourites,
+    getFavouritePokemonByName,
+  } = React.useContext(PokemonContext);
 
   const navigation = useNavigation<favouritePokemonProp>();
 
+  const { name } = pokemon;
+
   useEffect(() => {
     const getPokemons = async () => {
-      const isPokemonInStorage = await checkFavouritePokemon(name);
+      const isPokemonInStorage = await checkFavouritePokemon(pokemon);
```

Comment:
Wydaje mi się że już o tym pisałem w poprzednim PRku, ale to powinno być częścią kontekstu